design-toolkit/components/src/components/table/table-header-cell.tsx

```diff
@@ -10,20 +10,166 @@
+import { TableContext } from '.';
+import { Button } from '../button';
+import { Icon } from '../icon';
+import { Menu } from '../menu';
+import { TableStyles, headerCellStyles } from './styles';
+import { ColumnKebabMenuItems, type TableHeaderCellProps } from './types';
+
+const { headerCellButton, headerKebab, menuItem } = TableStyles();
 
 export function HeaderCell({
   ref,
   className,
   narrow,
+  header,
   ...props
-}: TableHeaderCellProps) {
+}: TableHeaderCellProps<any>) {
+  const {
+    enableColumnReordering,
+    enableSorting,
+    moveColumnLeft,
+    moveColumnRight,
+    persistHeaderKebabMenu,
+    setColumnSelection,
+  } = useContext(TableContext);
+
+  const [hoveredArrow, setHoveredArrow] = useState<boolean>(false);
+
+  const showKebab = enableColumnReordering || enableSorting;
+
+  const showHeaderIcon = (isAsc: boolean) => {
+    return (
+      <div
+        onMouseEnter={() => setHoveredArrow(true)}
+        onMouseLeave={() => setHoveredArrow(false)}
+      >
+        {hoveredArrow ? <Kebab /> : isAsc ? <ArrowUp /> : <ArrowDown />}
+      </div>
+    );
+  };
+
+  const tableMenu = (header: any) => {
+    if (!(enableSorting || enableColumnReordering)) {
+      return;
+    }
+    return (
+      <Menu.Trigger
+        onOpenChange={(isOpen) => {
+          if (isOpen) {
+            setColumnSelection?.(header.column.id);
+          } else {
+            setColumnSelection?.(null);
+          }
+        }}
+      >
+        <Button variant='icon' aria-label='Menu' className='p-s'>
+          <Icon>
+            {header?.column.getIsSorted() === 'asc' ? (
+              showHeaderIcon(true)
+            ) : header?.column.getIsSorted() === 'desc' ? (
+              showHeaderIcon(false)
+            ) : (
+              <div
+                className={headerKebab({
+                  persistKebab: persistHeaderKebabMenu,
+                })}
+              >
+                <Kebab />
+              </div>
+            )}
+          </Icon>
+        </Button>
+        <Menu>
+          {enableColumnReordering && (
+            <>
+              <Menu.Item
+                classNames={{ item: menuItem() }}
+                onAction={() => {
+                  moveColumnLeft
+                    ? moveColumnLeft(header.column.getIndex() ?? -1)
+                    : null;
+                }}
+                isDisabled={header.column.getIsFirstColumn('center')}
+              >
+                <Menu.Item.Label>{ColumnKebabMenuItems.Left}</Menu.Item.Label>
+              </Menu.Item>
+              <Menu.Item
+                classNames={{ item: menuItem() }}
+                onAction={() => {
+                  moveColumnRight
+                    ? moveColumnRight(header.column.getIndex() ?? -1)
+                    : null;
+                }}
+                isDisabled={header.column.getIsLastColumn('center')}
+              >
+                <Menu.Item.Label>{ColumnKebabMenuItems.Right}</Menu.Item.Label>
```

Comment:
Since you're only dealing with text that's only used once, and because the Menu.Item automatically wraps text children with the correct element, I would simplify this to be `<Menu.Item ... >Move Column Right</Menu.Item>`